src/etl/ReportingETL.cpp

```diff
@@ -912,6 +912,171 @@ ReportingETL::monitor()
+        auto const results = resolver.async_resolve(ip, port, yield[ec]);
+        if (ec)
+            return {};
+
+        BOOST_LOG_TRIVIAL(trace) << __func__ << "Connecting websocket";
+        // Make the connection on the IP address we get from a lookup
+        ws->next_layer().async_connect(results, yield[ec]);
+        if (ec)
+            return false;
+
+        BOOST_LOG_TRIVIAL(trace)
+            << __func__ << "Performing websocket handshake";
+        // Perform the websocket handshake
+        ws->async_handshake(ip, "/", yield[ec]);
+        if (ec)
+            return false;
+
+        std::optional<boost::json::value> marker;
+
+        BOOST_LOG_TRIVIAL(trace) << __func__ << "Sending request";
+        auto getRequest = [&](auto marker) {
+            boost::json::object request = {
+                {"command", "ledger_data"},
+                {"ledger_index", ledgerIndex},
+                {"binary", true},
+                {"out_of_order", true},
+                {"limit", 2048}};
+
+            if (marker)
+                request["marker"] = *marker;
+            return request;
+        };
+
+        do
+        {
+            // Send the message
+            ws->async_write(
+                net::buffer(boost::json::serialize(getRequest(marker))),
+                yield[ec]);
+            if (ec)
+            {
+                BOOST_LOG_TRIVIAL(error)
+                    << __func__ << "error writing = " << ec.message();
+                return false;
+            }
+
+            beast::flat_buffer buffer;
+            ws->async_read(buffer, yield[ec]);
+            if (ec)
+            {
+                BOOST_LOG_TRIVIAL(error)
+                    << __func__ << "error reading = " << ec.message();
+                return false;
+            }
+
+            auto raw = beast::buffers_to_string(buffer.data());
+            auto parsed = boost::json::parse(raw);
+
+            if (!parsed.is_object())
+            {
+                BOOST_LOG_TRIVIAL(error)
+                    << __func__ << "Error parsing response: " << raw;
+                return false;
+            }
+            BOOST_LOG_TRIVIAL(trace)
+                << __func__ << "Successfully parsed response " << parsed;
+
+            auto response = parsed.as_object();
+            if (response.contains("error"))
+            {
+                BOOST_LOG_TRIVIAL(error)
+                    << __func__ << "Response contains error: " << response;
+                auto err = response["error"];
+                if (err.is_string() && err.as_string() == "lgrNotFound")
+                {
+                    BOOST_LOG_TRIVIAL(warning)
+                        << __func__
+                        << " ledger not found. ledger = " << ledgerIndex
+                        << ". trying again";
+                    continue;
+                }
+                return false;
+            }
+            response = response["result"].as_object();
```

Comment:
consider using `auto const& result` instead

Reply:
well, now that you changed response to `auto&` you are rewriting your original response inside of `parsed` so this line needs to change i think? i'd have either what you had before or split into two differently named `auto const&` instead.